### PR TITLE
board: Add power-on reset env variable

### PR DESCRIPTION
**Pull request by rajaganeshr**
Adds support to read syscontrol reset status register to check if BMC booting after AC power on. Exports the Boolean value in u-boot env var, so that BMC can read DIMM SPD.

### CODE REVIEW

File: board/aspeed/evb_ast2700/evb_ast2700.c [hunk 1]
```diff
@@ -233,6 +235,23 @@ void aspeed_errata_app_note_6_pcie_ep()
...
+#define ASPEED_SYS_SCRATCH_7FC 0x12C027FC
+
+void update_por_env(void)
+{
+       const char *s;
+       u32 por_rst = readl(ASPEED_SYS_SCRATCH_7FC);
+
+       /* set reset reason env */
+       printf("Scratch register value: 0x%08x\n", por_rst);
+       if (por_rst & (u32)0x1)
```
Comment (mahkurap) [inline board/aspeed/evb_ast2700/evb_ast2700.c]: you may want to define a macro for the scratch register bit mask and use it here.

File: arch/arm/mach-aspeed/ast2700/cpu-info.c [hunk 2]
```diff
@@ -47,9 +46,9 @@ void ast2700_print_soc_id(void)
 #define SYS_SRST		BIT(0)
 
 #define WDT_RST_BIT_MASK(s)	(GENMASK(3, 0) << (s))
-#define BIT_WDT_SOC(s)		(BIT(0) << (s))
+#define BIT_WDT_SOC(s)		(BIT(2) << (s))
```
Comment (mahkurap) [inline arch/arm/mach-aspeed/ast2700/cpu-info.c:50]: how are these changes related to POR?

File: arch/arm/mach-aspeed/ast2700/cpu-info.c [hunk 3]
```diff
@@ -83,10 +82,13 @@ void ast2700_print_wdtrst_info(void)
 
 #define SYS_EXTRST		BIT(1)
 #define SYS_SRST		BIT(0)
+/* Sys Scratch reg SCU7FC */
+#define ASPEED_SYS_SCRATCH_7FC 0x12C027FC
 
 void ast2700_print_sysrst_info(void)
 {
 	u32 sys_rst = readl(ASPEED_CPU_RESET_LOG1);
+	writel(sys_rst, ASPEED_SYS_SCRATCH_7FC);
```
Comment (mahkurap) [inline arch/arm/mach-aspeed/ast2700/cpu-info.c:94]: why are we writing to 0x12C027FC?  You may want to add a sentence about this as a comment or in the commit log.

**Review by mahkurap — CHANGES_REQUESTED** [review]

File: arch/arm/mach-aspeed/ast2700/cpu-info.c [hunk 4]
```diff
@@ -47,9 +46,9 @@ void ast2700_print_soc_id(void)
 #define SYS_SRST		BIT(0)
 
 #define WDT_RST_BIT_MASK(s)	(GENMASK(3, 0) << (s))
-#define BIT_WDT_SOC(s)		(BIT(0) << (s))
+#define BIT_WDT_SOC(s)		(BIT(2) << (s))
 #define BIT_WDT_FULL(s)		(BIT(1) << (s))
-#define BIT_WDT_ARM(s)		(BIT(2) << (s))
+#define BIT_WDT_ARM(s)		(BIT(0) << (s))
 #define BIT_WDT_SW(s)		(BIT(3) << (s))
```
Comment (ojayanth) [inline arch/arm/mach-aspeed/ast2700/cpu-info.c:54]: Please make the above  two changes with cherry pick  https://github.com/AspeedTech-BMC/u-boot/commit/86500b3733c221e1287e86e80fb496704ba06873

**Review by ojayanth — CHANGES_REQUESTED** [review]

File: board/aspeed/evb_ast2700/evb_ast2700.c (hunk 1 above)
Comment (rajaganeshr) [reply board/aspeed/evb_ast2700/evb_ast2700.c]: Done

File: arch/arm/mach-aspeed/ast2700/cpu-info.c (hunk 2 above)
Comment (rajaganeshr) [reply arch/arm/mach-aspeed/ast2700/cpu-info.c:50]: Part of aspeed repo changes, cherrypicked origin commit in same pull request

File: arch/arm/mach-aspeed/ast2700/cpu-info.c (hunk 3 above)
Comment (rajaganeshr) [reply arch/arm/mach-aspeed/ast2700/cpu-info.c:94]: Done

File: arch/arm/mach-aspeed/ast2700/cpu-info.c (hunk 4 above)
Comment (rajaganeshr) [reply arch/arm/mach-aspeed/ast2700/cpu-info.c:54]: Done

File: board/aspeed/evb_ast2700/evb_ast2700.c [hunk 5]
```diff
@@ -233,6 +239,20 @@ void aspeed_errata_app_note_6_pcie_ep()
        printf("applying Aspeed app note 6 pcie ep ... Done\n");
 }
 
+void update_por_env(void)
```
Comment (ojayanth) [inline board/aspeed/evb_ast2700/evb_ast2700.c:242]: OpenBMC design document https://github.com/openbmc/docs/blob/master/designs/bmc-reboot-cause-update.md provides the current way of BMC reset handling. In this document "Each BMC vendor needs to record the results of the retrieved reboot cause to the specified path." . and PSM consumes this . Can you look the driver based approch helps to meet AMD requirement?

Comment (rajaganeshr) [reply board/aspeed/evb_ast2700/evb_ast2700.c:242]: The proposal mentions 
> "Each BMC vendor must provide a driver to retrieve the BMC reboot cause and record the result at the specified location."

I guess we can merge u-boot approach for now and later migrate to when Aspeed provides a driver.

**Review by mahkurap — APPROVED** [review]

**Review by supven01 — APPROVED** [review]

File: board/aspeed/evb_ast2700/evb_ast2700.c (hunk 5 above)
Comment (ojayanth) [reply board/aspeed/evb_ast2700/evb_ast2700.c:242]: As discussed , will use this approch for now , till the aspeed kernel support is available.

**Review by ojayanth — APPROVED** [review]
